color/src/temperature.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 the Color Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![expect(
+    clippy::excessive_precision,
+    reason = "We are sticking with the constants from Tanner Helland."
+)]
+
+use crate::{AlphaColor, ColorSpace, OpaqueColor};
+
+#[cfg(all(not(feature = "std"), not(test)))]
+use crate::floatfuncs::FloatFuncs;
+
+// Implementation notes:
+//
+// * These can't be const yet as `clamp`, `ln`, and `powf` are not `const`.
+
+impl<CS: ColorSpace> AlphaColor<CS> {
+    /// Convert a temperature in Kelvin to a color.
+    ///
+    /// The Kelvin temperature will be clamped to a range of 1000-40,000.
+    pub fn from_kelvin(kelvin: f32) -> Self {
+        let [r, g, b] = CS::from_linear_srgb(from_kelvin(kelvin));
+        Self::new([r, g, b, 1.])
+    }
+}
+
+impl<CS: ColorSpace> OpaqueColor<CS> {
+    /// Convert a temperature in Kelvin to a color.
+    ///
+    /// The Kelvin temperature will be clamped to a range of 1000-40,000.
+    pub fn from_kelvin(kelvin: f32) -> Self {
+        Self::new(CS::from_linear_srgb(from_kelvin(kelvin)))
+    }
+}
+
+/// Convert a Kelvin temperature into a linear RGB color.
```

Comment:
I believe this is actually calculating sRGB (i.e. non-linear), but would need to double check.